### DISCUSSION TITLE
Performance Analysis

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#2949] [Help]
How to identify the memory, cpu usage of imagemagick in linux terminal ?

**Comment by programmerarunkumar** [comment]
> How to identify the memory, cpu usage of imagemagick in linux terminal ?

Is there any way to analyse the imagemagick usage of  cpu , memory at runtime in a server , because server is  down for few times  when we calling the image magick code to resize the image

**Comment by urban-warrior** [comment]
See https://imagemagick.org/script/architecture.php.  You can set resource limits to prevent ImageMagick from thrashing your host, assuming that is the problem, see https://imagemagick.org/script/architecture.php#tera-pixel.  To enforce the limits system-wide, refer to https://imagemagick.org/script/security-policy.php.

**Reply by programmerarunkumar** [reply]
What Happen when number of thread accessing a imagemagick code is higher than the value of thread in policy.xml or identify -list resource ?

**Comment by snibgo** [comment]
One instance of an IM function will not create more threads than are shown in "identify -list resource". But you might create multiple instances of IM, or you might call an IM function from multiple threads. (See your "thread safe" post https://github.com/ImageMagick/ImageMagick/discussions/2931 )

This is generally not a good idea. For example, if your CPU can run 8 threads, then IM may try to use 8 threads. If you also create 8 threads that call IM functions, that would run 64 threads. This would create multiple thread-swapping, and an overhead in creating and unwinding threads, so expect poor performance.
